### DISCUSSION TITLE
Branch Protection rule to consider the "overall workflow status"

### DISCUSSION THREAD

**Discussion by nooblucker** [community/community#46978] [Repositories]
### Select Topic Area

Product Feedback

### Body

At the moment it is possible to configure which jobs must be successful before a pull request can be merged to a specific branch

![grafik](https://user-images.githubusercontent.com/3714349/218114108-2b5de308-01c4-45b1-975d-58237df6cef1.png)

When a new job is added to a workflow, the required status checks might need to be updated to prevent merging in case the newly added job fails. This requires manual maintenance of the required status checks in the branch protection rules and doesn't scale.If the required status checks aren't updated manually, the overall status of a PR workflow run can be "failed", while the PR can still be merged.We want an option to prevent merging of a PR, if the overall workflow status is "failed".

**Comment by tboddyspargo** [comment]
This would be quite valuable as a catch-all when dynamic configuration, optional jobs, or path filtering are used to include checks only in certain conditions. I expect it would also be compatible with situations where checks refer to 3rd party workflows/jobs (like CircleCI), which might also be dynamically included.
